routing/pathfind.go

```diff
@@ -189,6 +191,23 @@ func newRoute(sourceVertex route.Vertex,
 					"custom records")
 			}
 			customRecords = finalHop.records
+
+			// If we're attaching a payment addr but the receiver
+			// doesn't support both TLV and payment addrs, fail.
+			payAddr := supports(lnwire.PaymentAddrOptional)
+			if (!tlvPayload || !payAddr) &&
+				finalHop.paymentAddr != nil {
+
+				return nil, errors.New("cannot attach " +
+					"payment addr")
+			}
+
+			// Otherwise attach the mpp record if it exists.
+			if finalHop.paymentAddr != nil {
+				mpp = record.NewMPP(
+					finalHop.amt, *finalHop.paymentAddr,
+				)
+			}
```

Comment:
Should we do this here, or create `mpp` and `requiredFeatures` fields on `finalHopParams`? It is a question about how intelligent we want `newRoute` to be.